tests/testthat/test-helpers_design.R

```diff
@@ -205,3 +205,203 @@ test_that("h_determine_dlts returns correctly updated data object for placebo =
   expect_s4_class(result, "Data")
   expect_equal(result, expected_result)
 })
+
+test_that("simulate for the class design works end to end", {
+  emptydata <- Data(doseGrid = c(1, 3, 5, 10, 15, 20, 25, 40, 50, 80, 100))
+
+  model <- LogisticLogNormal(
+    mean = c(-0.85, 1),
+    cov =
+      matrix(c(1, -0.5, -0.5, 1),
+        nrow = 2
+      ),
+    ref_dose = 56
+  )
+
+  myNextBest <- NextBestNCRM(
+    target = c(0.2, 0.35),
+    overdose = c(0.35, 1),
+    max_overdose_prob = 0.25
+  )
+
+  mySize1 <- CohortSizeRange(
+    intervals = c(0, 30),
+    cohort_size = c(1, 3)
+  )
+  mySize2 <- CohortSizeDLT(
+    intervals = c(0, 1),
+    cohort_size = c(1, 3)
+  )
+  mySize <- maxSize(mySize1, mySize2)
+
+  myStopping1 <- StoppingMinCohorts(nCohorts = 3)
+  myStopping2 <- StoppingTargetProb(
+    target = c(0.2, 0.35),
+    prob = 0.5
+  )
+  myStopping3 <- StoppingMinPatients(nPatients = 20)
+  myStopping <- (myStopping1 & myStopping2) | myStopping3
+
+  myIncrements <- IncrementsRelative(
+    intervals = c(0, 20),
+    increments = c(1, 0.33)
+  )
+
+  design <- Design(
+    model = model,
+    nextBest = myNextBest,
+    stopping = myStopping,
+    increments = myIncrements,
+    cohort_size = mySize,
+    data = emptydata,
+    startingDose = 3
+  ) # check if default design constructor is enough to test simulate method
+
+  myTruth <- probFunction(model, alpha0 = 7, alpha1 = 8)
+
+  options <- McmcOptions(
+    burnin = 100,
+    step = 2,
+    samples = 1000,
+    rng_kind = "Mersenne-Twister",
+    rng_seed = 1234
+  )
+  time <- system.time(mySims <- simulate(design,
+    args = NULL,
+    truth = myTruth,
+    nsim = 1,
+    seed = 819,
+    mcmcOptions = options,
+    parallel = FALSE,
+    derive = list(
+      max_mtd = max,
+      mean_mtd = mean,
+      median_mtd = median
+    ),
+  ))[3]
+
+  expected <-
```

Comment:
this is too detailed / too much maintenance when numbers change etc.

Reply:
Thanks for guidance! I will adopt.

Reply:
@danielinteractive Hi Daniel, I put the new "end-to-end" test for simulate in "test-helpers_design.R". When moving it over to "test-Design-methods.R" I realized that there are also tests (using snapshots) for simulate already implemented. Those use fixed rng seeds, therefore "mimicking" quite closely what you suggested not to do. I don´t know what to keep and what to get rid of there?!

![image](https://github.com/Roche/crmPack/assets/97281825/3a959a87-689b-4b0f-b203-68ec719a5c6d)

![image](https://github.com/Roche/crmPack/assets/97281825/965668d8-7cfd-4f91-b9e9-5c70ee270504)

Reply:
I would just leave existing tests. More coverage does not hurt us